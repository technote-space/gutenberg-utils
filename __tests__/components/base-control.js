import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe( 'BaseControl', () => {
	let label;
	beforeEach( () => {
		label = wp.components.BaseControl.VisualLabel;
	} );

	afterEach( () => {
		wp.components.BaseControl.VisualLabel = label;
		jest.resetModules();
	} );

	it( 'should have VisualLabel 1', () => {
		wp.components.BaseControl.VisualLabel = () => {
		};
		const control = require( '../../src/components/base-control' ).default;
		expect( control ).toHaveProperty( 'VisualLabel' );
	} );

	it( 'should have VisualLabel 2', () => {
		delete wp.components.BaseControl.VisualLabel;
		const control = require( '../../src/components/base-control' ).default;
		expect( control ).toHaveProperty( 'VisualLabel' );
	} );

	it( 'should render VisualLabel', () => {
		delete wp.components.BaseControl.VisualLabel;
		const BaseControl = require( '../../src/components/base-control' ).default;
		const wrapper = shallow(
			<BaseControl.VisualLabel className='test-label'>
				test label
			</BaseControl.VisualLabel>,
		);
		expect( toJson( wrapper ) ).toMatchSnapshot();
		expect( wrapper.hasClass( 'test-label' ) ).toBeTruthy();
		expect( wrapper.text() ).toBe( 'test label' );
	} );
} );
