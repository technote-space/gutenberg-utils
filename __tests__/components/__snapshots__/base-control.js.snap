// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseControl should render VisualLabel 1`] = `
<span
  className="components-base-control__label test-label"
>
  test label
</span>
`;
